pkg/api/controllers/samlconfig/samlconfig.go

```diff
@@ -0,0 +1,33 @@
+package samlconfig
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/auth/providers/saml"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+)
+
+func Register(apiContext *config.ScaledContext) {
+	apiContext.Management.AuthConfigs("").AddHandler("authConfigController", sync)
+}
+
+func sync(key string, config *v3.AuthConfig) error {
+	samlConfig := &v3.SamlConfig{}
+	if key == "" || config == nil {
+		return nil
+	}
+
+	if samlConfStr, ok := config.Annotations["tempSamlConfigbyte"]; ok {
```

Comment:
I thought we were going to drop the annotation and just do this based off the config itself. I think I understand your problem now though: the type AuthCOnfig is hiding the saml specific fields. I think you can use this paradigm to get and update the specific type:
get: https://github.com/rancher/rancher/blob/4d560ca5a8e5391a62fda1e0a00e334fb6153c73/pkg/auth/providers/github/github_provider.go#L66-L69

update: 
https://github.com/rancher/rancher/blob/4d560ca5a8e5391a62fda1e0a00e334fb6153c73/pkg/auth/providers/github/github_provider.go#L92